[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/miguepoloc/toma-decisiones-mcda/blob/main/05_promethee_cacao.ipynb)

# PROMETHEE II, caso cacao

Mismo caso. Contenido completo en la Sesión 4 (ELECTRE/PROMETHEE) del curso.

**Detalle importante, verificado contra el código fuente de `promethee_ii`:** a diferencia de `topsis_method`/`vikor_method`, esta función NO recibe un parámetro de tipo beneficio/costo, asume que "mayor es mejor" en todas las columnas. Los criterios de costo (Humedad, Conductividad) se **niegan** antes de llamar la función.

In [1]:
!pip install -q pyDecision


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: pip install --upgrade pip


In [2]:
import numpy as np
from pyDecision.algorithm import promethee_ii

zonas = ["Bonda", "Guachaca", "San Pedro", "Palmor"]
dataset = np.array([
    [27, 78, 6.2, 0.9],
    [26, 85, 5.8, 1.5],
    [24, 82, 6.5, 0.7],
    [22, 88, 5.5, 1.2],
], dtype=float)

dataset_orientado = dataset.copy()
dataset_orientado[:, 1] = -dataset_orientado[:, 1]  # Humedad es costo
dataset_orientado[:, 3] = -dataset_orientado[:, 3]  # Conductividad es costo

rangos = dataset.max(axis=0) - dataset.min(axis=0)
pesos = [0.25, 0.30, 0.25, 0.20]
Q = [0, 0, 0, 0]
S = [0, 0, 0, 0]
P = list(rangos)
F = ['t3', 't3', 't3', 't3']  # función lineal simple

## Flujos y ranking

In [3]:
flujo = promethee_ii(dataset_orientado, W=pesos, Q=Q, S=S, P=P, F=F,
                      sort=False, topn=0, graph=False, verbose=False)

print("Ranking final PROMETHEE II:")
for z, fila in sorted(zip(zonas, flujo), key=lambda x: -x[1][1]):
    print(f"  {z:12s} phi_neto = {fila[1]:+.3f}")

Ranking final PROMETHEE II:
  Bonda        phi_neto = +0.485
  San Pedro    phi_neto = +0.292
  Guachaca     phi_neto = -0.195
  Palmor       phi_neto = -0.582


**Resultado esperado** (coincide con lo publicado en la Sesión 4 del curso): 1º Bonda (+0.485) · 2º San Pedro (+0.292) · 3º Guachaca (-0.195) · 4º Palmor (-0.582), idéntico a AHP y VIKOR.